### PR TITLE
Set YUI REST API parameters for ssh installations on powerVM and zkvm

### PR DESCRIPTION
**Pull request by rwx788**
Libyui REST API linuxrc parameters we specify in the bootloader
are not passed to the installer, when it's started in the ssh, so we
should specify them in the same way, as we do when start YaST modules in
the installed system.

See [poo#90776](https://progress.opensuse.org/issues/90776).

[Verification runs](https://openqa.suse.de/tests/overview?distri=sle&build=rwx788%2Fos-autoinst-distri-opensuse%23pvm&version=15-SP3).

### CODE REVIEW

File: lib/YuiRestClient.pm [hunk 1]
```diff
@@ -65,7 +70,7 @@ sub setup_libyui_running_system {
     my $host = get_var('YUI_SERVER');
     record_info('PORT',   "Used port for libyui: $port");
     record_info('SERVER', "Connecting to: $host");
-    set_var('YUI_PARAMS', "YUI_HTTP_PORT=$port YUI_HTTP_REMOTE=1 YUI_REUSE_PORT=1");
+    set_var('YUI_PARAMS', get_yui_params_string());
```
Comment (jknphy) [inline lib/YuiRestClient.pm:73]: could we pass the port instead of getting the variable twice?
Also we could use it below in `sub set_libyui_backend_vars`

**Review by jknphy — COMMENTED** [review]
I noticed in commit message:
```
Besides explicit ssh installations, we also get to the ssh when s/perfrom/perform/
textmode installation on powerVM, zVM and zKVM. So this code will be
s/helpfull/helpful/ in those cases too.
```

File: lib/YuiRestClient.pm (hunk 1 above)
Comment (rwx788) [reply lib/YuiRestClient.pm:73]: Makes sense, I will use `get_required_var` here than, as it should be set by this point of time.

File: tests/installation/setup_libyui.pm [hunk 2]
```diff
@@ -54,7 +54,15 @@ sub run {
                 return $+{ip} if ($ip =~ $ip_regexp);
         });
         set_var('YUI_SERVER', $ip);
+    } elsif (is_ssh_installation) {
+        my $cmd = "";
```
[the remark this answers is not in the document]
Comment (rwx788) [reply tests/installation/setup_libyui.pm]: changed ;)

**Comment by rwx788** [comment]
> I noticed in commit message:
> 
> ```
> Besides explicit ssh installations, we also get to the ssh when s/perfrom/perform/
> textmode installation on powerVM, zVM and zKVM. So this code will be
> s/helpfull/helpful/ in those cases too.
> ```

Nice catch, thanks!

**Review by jknphy — APPROVED** [review]

File: tests/installation/bootloader_zkvm.pm [hunk 3]
```diff
@@ -81,7 +81,11 @@ sub run {
         if (check_var("VIDEOMODE", "text")) {
             wait_serial("run 'yast.ssh'", 300) || die "linuxrc didn't finish";
             select_console("installation");
-            enter_cmd("TERM=linux yast.ssh") && record_soft_failure('bsc#1054448');
+            # If libyui REST API is used, we set it up in installation/setup_libyui
+            unless (get_var('YUI_REST_API')) {
+                enter_cmd("TERM=linux yast.ssh");
+                record_soft_failure('bsc#1054448');
+            }
```
Comment (alvarocarvajald) [inline tests/installation/bootloader_zkvm.pm:88]: Seems this has introduced a soft failure for all zkvm tests where **YUI_REST_API** is not set, for example:

https://openqa.suse.de/tests/5899471 vs https://openqa.suse.de/tests/5926007

Seeing as bsc#1054448 has been closed for a while, would it be acceptable to drop this `record_soft_failure`?

Comment (jknphy) [reply tests/installation/bootloader_zkvm.pm:88]: Thanks Alvaro, we were not aware, #12519.

Comment (alvarocarvajald) [reply tests/installation/bootloader_zkvm.pm:88]: Thanks Joaquin.

**Pull request merged**